src/internal/easee/controller.go

```diff
@@ -97,3 +113,64 @@ func (c *controller) ElectricityMeterReport(unit string) (float64, error) {
 		return 0, errors.Errorf("unsupported unit: %s", unit)
 	}
 }
+
+func (c *controller) SetChargepointChargingMode(mode string) error {
+	switch mode {
+	case ChargingModeSlow:
+		return c.client.SetChargingCurrent(c.chargerID, SlowChargingCurrentInAmpers)
+	case ChargingModeNormal:
+		return c.client.SetChargingCurrent(c.chargerID, NormalChargingCurrentInAmpers)
+	default:
+		return errors.Errorf("unsupported mode: %s", mode)
+	}
+}
+
+func (c *controller) ChargepointChargingModeReport() (string, error) {
+	state, err := c.client.ChargerState(c.chargerID)
+	if err != nil {
+		return "", fmt.Errorf("failed to fetch charger state for ID: %s: %w", c.chargerID, err)
+	}
+
+	if state.ChargerOpMode.String() != ChargerModeCharging {
+		return "", fmt.Errorf("cannot report charging mode if device is not charging")
```

Comment:
Nope, this should be persisted. Charging mode should be saved in adapter persistent storage the first time it's being set. If it is not set by default it should be normal. Then you should be setting it on each started session, right?

Reply:
Outdated.